src/views/Character/Character.jsx

```diff
@@ -0,0 +1,33 @@
+import React, { useEffect, useState } from 'react';
+import { useParams, useHistory } from 'react-router-dom';
+import CharacterDetails from '../../components/CharacterDetails/CharacterDetails';
+import { fecthCharacterbyId } from '../../services/characters';
+import './Character.css';
+
+export default function Character() {
+  const [details, setDetails] = useState({});
+  const { characterId } = useParams();
+  const history = useHistory();
+
+  useEffect(() => {
+    const getDetails = async () => {
+      const resp = await fecthCharacterbyId(characterId);
+      setDetails(resp);
+    };
+    getDetails();
+  }, [characterId]);
+
+  const handleClick = (e) => {
+    e.preventDefault();
+    history.push('/');
+  };
+
+  const { name, imageUrl, url } = details;
+
+  return (
+    <div>
+      <button onClick={handleClick}>Back</button>
+      <CharacterDetails name={name} imageUrl={imageUrl} url={url} />
```

Comment:
We could just spread in details here instead of de-structing on line 25. `details={...details}`